components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/cache/AuthenticationResultCache.java

```diff
@@ -93,14 +95,33 @@ public void addToCache(AuthenticationResultCacheKey key, AuthenticationResultCac
      * @return Cached entry.
      */
     public AuthenticationResultCacheEntry getValueFromCache(AuthenticationResultCacheKey key) {
+
         AuthenticationResultCacheEntry entry = super.getValueFromCache(key);
         if (entry == null && isTemporarySessionDataPersistEnabled) {
             entry = (AuthenticationResultCacheEntry) SessionDataStore.getInstance().
                     getSessionData(key.getResultId(), CACHE_NAME);
+            if (entry != null && isCacheEntryExpired(entry)) {
+                return null;
+            }
         }
         return entry;
     }
 
+    private boolean isCacheEntryExpired(AuthenticationResultCacheEntry entry) {
+
+        String createdTimestamp = entry.getResult().getProperty(FrameworkConstants.CREATED_TIMESTAMP).toString();
+        if (createdTimestamp != null) {
+            if (FrameworkUtils.getCurrentStandardNano() >
+                    entry.getValidityPeriod() + Long.parseLong(createdTimestamp) * 1000000) {
+                if (log.isDebugEnabled()) {
```

Comment:
We can skip this check since we don't have any expensive string operations.

Reply:
Fixed in https://github.com/wso2/carbon-identity-framework/pull/6330/commits/b0dbb60782e2b634e1fc460c74f3e391da75f3b0